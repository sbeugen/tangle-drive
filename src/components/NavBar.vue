<template>
  <ul>
    <!-- <li @click="aboutClickHandler">AboutMe</li> -->
    <li @click="gitHubClickHandler">GitHub</li>
  </ul>
</template>

<script>
  export default {
    methods: {
      aboutClickHandler() {
        this.$router.push('about')
      },
      gitHubClickHandler() {
        window.open('https://github.com/sbeugen/tangle-drive', '_blank')
      }
    }
  }
</script>

<style scoped>
  ul > li {
    list-style: none;
    padding: 0;
    float: right;
    margin: 15px 10px;
    font-size: 20px;
  }
  li {
    cursor: pointer;
  }
  li:hover {
    color: lightgrey;
  }
</style>
