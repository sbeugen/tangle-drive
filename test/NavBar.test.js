import { shallow} from 'vue-test-utils'
import NavBar from '../src/components/NavBar'

describe('NavBar.test.js', () => {
  let cmp
  const $router = {
    push: jest.fn()
  }

  beforeEach(() => {
    cmp = shallow(NavBar)
  })

  it('has expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })

  it('gitHubClickHandler is called on click on the github li', () => { 
    cmp.vm.gitHubClickHandler = jest.fn()

    cmp.find('li').trigger('click')
    expect(cmp.vm.gitHubClickHandler).toBeCalled()
  })

  it('window.open is calles in the gitHubClickHandler', () => {
    window.open = jest.fn()

    cmp.find('li').trigger('click')
    expect(window.open).toBeCalled()
  })
})

