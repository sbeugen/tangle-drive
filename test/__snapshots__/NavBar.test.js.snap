// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar.test.js has expected html structure 1`] = `
<ul>
  <li>
    GitHub
  </li>
</ul>
`;
